Create .editorconfig

mdrmike opened the pull request:
@barryclark awesome you committed #39

However, regarding "passing on [editorconfig](http://editorconfig.org)," I'd urge you to reconsider.  

It's a newer standard, but virtually all active and recent projects are using it, [here's a list](https://github.com/editorconfig/editorconfig/wiki/Projects-Using-EditorConfig).  I'd consider it "best practice" and a signature of an active project.  It allows contributors to easily conform to the standards of the project -- and shouldn't require any interaction by the end user.

I've modified the original patch from @LeoColomb  to create this pull request.  I also reviewed the editorconfig from octopress -- and found it a bit too specific/redundant.  We might want to review the editorconfig for jekyll-now, but I think this is a good starting point.

This file is based on documented and simplified versions of [Jekyll-now patch](https://github.com/barryclark/jekyll-now/commit/6bdf909a90cf94f93a4ca2b82f2319fa6fdc29a6) from @LeoColomb and a simplified [Octopress editorconfig](https://github.com/imathis/octopress/commit/05f54ab16b167656fb58b1f6503296dd44518132?diff=unified)

mdrmike committed:
Create .editorconfig

Based on documented and simplified versions of
[Jekyll-now patch](https://github.com/barryclark/jekyll-now/commit/6bdf909a90cf94f93a4ca2b82f2319fa6fdc29a6)
and a simplified [Octopress editorconfig](https://github.com/imathis/octopress/commit/05f54ab16b167656fb58b1f6503296dd44518132?diff=unified)

barryclark commented:
@mediadoneright Thanks for putting this together! 

Let me have more of a think about it and I'll get back to you :)

mdrmike committed:
Update editorconfig to match Jekyll
There is [a pull-request for the Jekyll project](https://github.com/jekyll/jekyll/pull/3527) for a similar editorconfig file.  This change syncronizes the files.
